### PR TITLE
Extend RelatedEntitiesCard usages to allow custom columns

### PR DESCRIPTION
**Pull request by blakestoddard**
## Hey, I just made a Pull Request!

The `Depends on [components/resources]` tables on the entity detail page have been a source of questions from users for us because a) the naming (component/resource) doesn't match how we talk internally, and b) they showed columns that weren't entirely relevant to how we wanted to use them (in our case for resources as an example, we don't care about owner or lifecycle, but we do care about a custom compliance attribute). This allows you to specify a set of `TableColumns` via component props rather than needing to rebuild the component to support that.

This is a replacement PR for #15216 due to GH not letting you re-open a PR after a force push.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/wild-ads-pull.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-api-docs': patch
+'@backstage/plugin-catalog': patch
```
Comment (jhaals) [inline .changeset/wild-ads-pull.md:3]: plugin-catalog is at version 1.x so new features should be `minor`. I suggest you split this up into two changeset, they can contain the same text but one being patch and the other minor.

File: .changeset/wild-ads-pull.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+---
+
+Add a columns prop to certain components that use the EntityTable for easier extensibility.
```
Comment (jhaals) [inline .changeset/wild-ads-pull.md:6]:
```suggestion
Add a `columns` prop to certain components that use the EntityTable for easier extensibility.
```

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for the PR!

One `minor` thing with the changesets but it looks good to me otherwise.

**Review by freben — APPROVED** [review]

File: .changeset/wild-ads-pull.md (hunk 2 above)
Comment (blakestoddard) [reply .changeset/wild-ads-pull.md]: Adjusted in c9a9f3c

File: .changeset/wild-ads-pull.md (hunk 1 above)
Comment (blakestoddard) [reply .changeset/wild-ads-pull.md]: Sounds good, thanks Johan!

**Review by jhaals — APPROVED** [review]

**Pull request merged**